pkg/logger/progress_logger_test.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2022 VMware, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logger
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"testing"
+	"time"
+
+	regv1 "github.com/google/go-containerregistry/pkg/v1"
+)
+
+func TestProgressBar(t *testing.T) {
+	var regexProgress = regexp.MustCompile(`[0-9]*\.*[0-9]*\s[a-zA-Z]+\s/\s[0-9]*[\.]*[0-9]*\s[a-zA-Z]+\s\[-+>*\]\s[0-9]*\.[0-9]+%.*`)
+	tests := []struct {
+		name      string
+		regObject []regv1.Update
+	}{{
+		name:      "ProgressLogger",
+		regObject: []regv1.Update{{Total: 100, Complete: 100, Error: nil}},
+	}, {
+		name:      "ProgressLogger with Error",
+		regObject: []regv1.Update{{Total: 100, Complete: 20, Error: fmt.Errorf("test Error")}, {Total: 100, Complete: 100, Error: nil}},
```

Comment:
add test cases for 
- when total is set to 0 , it should print no progress bar
- when context is closed before End is called, progress bar should stop mid way and exit.